### PR TITLE
Fix runtime restore bug (inconsistency with NuGet 3)

### PR DESCRIPTION
**Pull request by anurse**
This should be the issue making us inconsistent with NuGet 3. Not sure if it will fix the clrcompression issue, but if it doesn't we can debug further.

/cc @davidfowl @BrennanConroy @muratg

### CODE REVIEW

File: src/Microsoft.Dnx.Tooling/Restore/RestoreOperations.cs [hunk 1]
```diff
@@ -26,6 +26,45 @@ public RestoreOperations(IReport report)
             var sw = new Stopwatch();
             sw.Start();
 
+            var dependencies = new List<LibraryDependency>();
+            if (context.RuntimeSpecs != null && !string.IsNullOrEmpty(context.RuntimeName))
+            {
+                // HACK(davidfowl): This is making runtime.json support package redirects
+
+                // Look up any additional dependencies for this package
+                var runtimeDependencies = context.RuntimeSpecs
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Tooling/Restore/RestoreOperations.cs]: What is this?

Comment (anurse) [inline src/Microsoft.Dnx.Tooling/Restore/RestoreOperations.cs]: You tell me, it's your code :)

Comment (anurse) [inline src/Microsoft.Dnx.Tooling/Restore/RestoreOperations.cs]: Ah, the LINQ. It's the equivalent of https://github.com/NuGet/NuGet3/blob/dev/src/NuGet.RuntimeModel/RuntimeGraph.cs#L147 (which is what the NuGet3 code called).

File: src/Microsoft.Dnx.Tooling/Restore/RestoreOperations.cs [hunk 2]
```diff
@@ -46,8 +87,9 @@ public RestoreOperations(IReport report)
                     }
                 }
 
+                dependencies.AddRange(node.Item.Dependencies);
+
                 var tasks = new List<Task<GraphNode>>();
-                var dependencies = node.Item.Dependencies ?? Enumerable.Empty<LibraryDependency>();
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Tooling/Restore/RestoreOperations.cs]: Not enough old code is removed. It's still evaluating runtime specs for each dependency and that's unnecessary.

Comment (anurse) [inline src/Microsoft.Dnx.Tooling/Restore/RestoreOperations.cs]: Yeah, I think my update failed to push last night ;)

**Comment by davidfowl** [comment]
:shipit:

**Pull request closed without merging**